### PR TITLE
Support for architecture in Visual Studio IDE

### PR DESCRIPTION
**Pull request by lacasseio**
### Context

See https://github.com/gradle/gradle-native/issues/916

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Check&tab=projectOverview&branch_Gradle_Check=lacasseio%2Fnative%2Fvisual-studio)
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by big-guy** [comment]
@lacasseio this needs to be based on release

File: subprojects/language-native/src/testFixtures/groovy/org/gradle/language/VariantContext.groovy [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        def dimensions = GroovyCollections.combinations(listOfDimensionArrays)
+        dimensions.each {
+            def dimension = new LinkedHashMap<String, VariantDimension>()
+            it.eachWithIndex { item, idx ->
+                if (item instanceof VariantDimension) {
+                    assert listOfDimensionArrays[idx] instanceof Iterable
+                    if (listOfDimensionArrays[idx].size() > 1) {
+                        dimension.put(item.dimensionName, item)
+                    }
+                } else if (item instanceof Map) {
```
[the remark this answers is not in the document]
Comment (lacasseio) [reply subprojects/language-native/src/testFixtures/groovy/org/gradle/language/VariantContext.groovy]: It's used in the publishing test where it passes the target machines as a list of `Map<String, String>` with `architecture` and `os` as the keys. I fixed the declared type which was wrong and doesn't know why it wasn't triggering a failure.

[the remark this answers is not in the document]
Comment (ghale) [reply subprojects/language-native/src/testFixtures/groovy/org/gradle/language/VariantContext.groovy]: Ah, ok.

File: subprojects/ide-native/src/integTest/groovy/org/gradle/ide/visualstudio/AbstractVisualStudioProjectIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            allprojects {
+                apply plugin: 'visual-studio'
+            }
+        """
+        settingsFile << """
+            rootProject.name = "${rootProjectName}"
+        """
+    }
+
+    def "ignores target machine not buildable from project configuration dimensions"() {
```
[the remark this answers is not in the document]
Comment (ghale) [reply subprojects/ide-native/src/integTest/groovy/org/gradle/ide/visualstudio/AbstractVisualStudioProjectIntegrationTest.groovy]: We can ignore it for now and answer the question with that issue - I think there's perhaps an argument that "build" should fail, but IDE metadata generation should still succeed.

**Pull request merged**